google/cloud/security/common/gcp_api/storage.py

```diff
@@ -101,3 +103,28 @@ def get_text_file(self, full_bucket_path):
             LOGGER.error('Unable to download file: %s', http_error)
             raise http_error
         return file_content
+
+    def get_buckets(self, project_id):
+        """Gets all GCS buckets for a project.
+
+        Args:
+            project_id: The project id for a GCP project.
+
+        Returns:
+            {
+              "kind": "storage#buckets",
+              "nextPageToken": string,
+              "items": [
+                buckets Resource
+              ]
+            }
+        """
+        storage_service_api = self.service.buckets()
+        try:
+            buckets_request = storage_service_api.list(project=project_id)
+            buckets = buckets_request.execute()
+            return buckets
+        except (HttpError, HttpLib2Error) as e:
+            LOGGER.error(api_errors.ApiExecutionError(project_id, e))
```

Comment:
If this error is just logged and not raised, we will not be able to tag the pipeline status in the inventory_loader.py, which is used for email notification.
```
run_statuses = _run_pipelines(pipelines)
```
https://github.com/GoogleCloudPlatform/forseti-security/blob/master/google/cloud/security/inventory/inventory_loader.py#L327